src/providers/ipa/ipa_common.c

```diff
@@ -81,9 +81,9 @@ int ipa_get_options(TALLOC_CTX *memctx,
     if (ipa_hostname == NULL) {
         ret = gethostname(hostname, sizeof(hostname));
         if (ret != EOK) {
+            ret = errno;
             DEBUG(SSSDBG_CRIT_FAILURE, "gethostname failed [%d][%s].\n", errno,
                       strerror(errno));
```

Comment:
Please, update also the `DEBUG` statement. Use `ret` here instead of `errno`.

Reply:
Thank you for the review.
Updated DEBUG statement to use ret instead of errno.